src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+
+const updatePeople = (loadPeople: Person[]) => {
+  const updatedPeople = loadPeople.map(person => {
+    return {
+      ...person,
+      mother:
+        loadPeople.find(mother => mother.name === person.motherName),
+      father:
+        loadPeople.find(father => father.name === person.fatherName),
+    };
+  });
+
+  return updatedPeople;
+};
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[] | null>(null);
+  const [hasError, setHasError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const { slug = '' } = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    const loadData = async () => {
+      try {
+        const loadPeople = await getPeople();
+
+        setPeople(updatePeople(loadPeople));
+        setIsLoading(false);
```

Comment:
to be exact even if error occurs isLoading should become false so do it after try catch or in finally block